Files/Helpers/AdaptiveLayoutHelpers.cs

```diff
@@ -2,19 +2,34 @@
 using Files.ViewModels.Previews;
 using System;
 using System.Linq;
+using System.Threading.Tasks;
 using Windows.Storage;
 
 namespace Files.Helpers
 {
     public static class AdaptiveLayoutHelpers
     {
-        public static bool PredictLayoutMode(FolderSettingsViewModel folderSettings, ItemViewModel filesystemViewModel)
+        public static async Task<bool> PredictLayoutMode(FolderSettingsViewModel folderSettings, ItemViewModel filesystemViewModel)
         {
             if (App.AppSettings.AreLayoutPreferencesPerFolder && App.AppSettings.AdaptiveLayoutEnabled && !folderSettings.LayoutPreference.IsAdaptiveLayoutOverridden)
             {
                 bool desktopIniFound = false;
 
-                var iniPath = System.IO.Path.Combine(filesystemViewModel.CurrentFolder.ItemPath, "desktop.ini");
+                string path = filesystemViewModel?.WorkingDirectory;
+
+                if (string.IsNullOrWhiteSpace(path))
+                {
+                    return false;
+                }    
+                else
+                {
+                    if (!(await StorageItemHelpers.ToStorageItem<StorageFolder>(path) is StorageFolder folder) || folder == null)
```

Comment:
Should not be needed. `ReadStringFromFile` does not crash if the file does not exist. Appcenter report the crash on line #17 so the change above is enough.

Reply:
But we are then checking for files and folders in this directory. Having this check allows us to skip the rest if the folder is invalid

@gave92

Reply:
This check with storagefiles is probably slower than doing the rest of the method :) Plus I'm not sure if it works for hidden folders

Reply:
Ok :)